estimate_covid_infections_from_deaths_data.R

```diff
@@ -0,0 +1,155 @@
+onset_death_params <- epiparameter::get_parameters(onset_to_death_dist)
+onset_death_max_days <- round(quantile(onset_to_death_dist, 0.999))  # Upper 99.9% range
+
+# Create an EpiNow2 LogNormal object for the onset-to-death distribution
+onset_death_lognormal <- EpiNow2::LogNormal(
+  meanlog = onset_death_params[["meanlog"]],
+  sdlog = onset_death_params[["sdlog"]],
+  max = onset_death_max_days
+)
+
+# Use EpiNow2 to convolve the infection-to-onset and onset-to-death distributions
+infection_to_death_dist <- incubation_lognormal + onset_death_lognormal
+
+# Extract the serial interval distribution
+serial_interval_dist <- epiparameter::epiparameter_db(
+  disease = "covid",
+  epi_name = "serial",
+  single_epiparameter = TRUE
+)
+
+# Display the serial interval distribution information
+serial_interval_dist
+
+# Find the upper 99.9% range for the serial interval
+serial_interval_max_days <- round(quantile(serial_interval_dist, 0.999))
+
+# Extract parameters for the serial interval
+serial_interval_params <- epiparameter::get_parameters(serial_interval_dist)
+
+# Create a LogNormal object for the serial interval
+serial_interval_lognormal <- EpiNow2::LogNormal(
+  meanlog = serial_interval_params[["meanlog"]],
+  sdlog = serial_interval_params[["sdlog"]],
+  max = serial_interval_max_days
+)
+
+# ============================================================================== #
+# ESTIMATE INFECTIONS AND VISUALIZE RESULTS
+# ============================================================================== #
+
+# Estimate infections using the non-mechanistic model
+infection_estimates <- EpiNow2::epinow(
+  data = uk_covid_deaths,  # Time series data
+  generation_time = EpiNow2::generation_time_opts(serial_interval_lognormal),  # Generation time
+  delays = EpiNow2::delay_opts(infection_to_death_dist),  # Delay from infection to death
+  rt = NULL  # Rt is not estimated
+)
+
+# Extract and filter infection estimates from the model output
+estimated_infections <- infection_estimates$estimates$summarised %>%
+  dplyr::filter(variable == "infections")
+
+# Annotate the infections plot with key interventions in the UK
+infection_estimates$plots$infections +
+  geom_vline(aes(xintercept = as.Date("2020-03-16")), linetype = 3) +
+  geom_text(
+    aes(
+      x = as.Date("2020-03-16"),
+      y = 3000,
+      label = "Non-essential contact advice"
+    ),
+    hjust = 0
+  ) +
+  geom_vline(aes(xintercept = as.Date("2020-03-23")), linetype = 3) +
+  geom_text(
+    aes(
+      x = as.Date("2020-03-23"),
+      y = 2500,
+      label = "Stay-at-home order (i.e. lockdown)"
+    ),
+    hjust = 0
+  ) +
+  labs(
+    title = "Estimated Dynamics of SARS-CoV-2 Infections in the UK",
+    subtitle = "Reconstructed using data on reported deaths. Dashed lines indicate key intervention dates."
+  )
+
+#' Further exploration
+
+#' - Explore the following priors on the Gaussian process and observe what changes in the output:
```

Comment:
would you like to specify the function where to explore?
```suggestion
#' - In EpiNow2::epinow(), explore the following priors on the Gaussian process and observe what changes in the output:
```